packages/react-utilities/src/compose/getSlots.ts

```diff
@@ -44,7 +47,7 @@ export function getSlots<SlotProps extends SlotPropsRecord = {}>(state: Componen
   type Slots = { [K in keyof SlotProps]: React.ElementType<SlotProps[K]> };
 
   const slots = ({
-    root: getSlot(typedState.components ? typedState.components.root : undefined, typedState.as),
+    root: getRootSlot(typedState.components ? typedState.components.root : undefined, typedState.as),
```

Comment:
```suggestion
    root: typedState.as || typedState.components?.root || 'div',
```

Do we really need `getRootSlot`?

Reply:
This is not equivalent, it fail on the case of root as a function